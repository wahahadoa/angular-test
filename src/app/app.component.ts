import { Component } from '@angular/core';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.css']
})
export class AppComponent {
  title = 'app';
  songs = [
    {
      title: 'Song 1',
      likes: 25
    },
    {
      title: 'Song 5',
      likes: 50
    },
    {
      title: 'Song 10',
      likes: 10
    }
  ];
  term = 5;
  changeMaker(value){
    let pennies = value * 100;
    const quarters = Math.floor(pennies / 25);
    pennies -= quarters * 25;
    const dimes = Math.floor(pennies / 10);
    pennies -= dimes * 10;
    const nickels = Math.floor(pennies / 5);
    pennies -= nickels * 5;
    pennies++;
    return {
      quarters,
      dimes,
      nickels,
      pennies,
    };
  }
}
