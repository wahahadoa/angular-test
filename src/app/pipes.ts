import { Pipe, PipeTransform } from '@angular/core';

@Pipe({
    name: 'filter',
    pure: false
})
export class FilterPipe implements PipeTransform {
    transform(items: any[], term): any {
        console.log('term', term);
      
        return term 
            ? items.filter(item => item.title.indexOf(term) !== -1)
            : items;
    }
}

@Pipe({
    name: 'sortBy'
})
export class SortByPipe implements PipeTransform {
    transform(items: any[], sortedBy: string): any {
        console.log('sortedBy', sortedBy);
        
        return items.sort((a, b) => {return b[sortedBy] - a[sortedBy]});
    }
}