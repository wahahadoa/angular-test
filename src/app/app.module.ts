import { BrowserModule } from '@angular/platform-browser';
import { NgModule } from '@angular/core';
import { FormsModule } from '@angular/forms';

import { AppComponent } from './app.component';
import {FilterPipe, SortByPipe} from './pipes'

@NgModule({
  declarations: [
    AppComponent,
    FilterPipe, 
    SortByPipe
  ],
  imports: [
    BrowserModule,
    FormsModule
  ],
  providers: [],
  bootstrap: [AppComponent]
})
export class AppModule { }
