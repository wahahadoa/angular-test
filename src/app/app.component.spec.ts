import { TestBed, async } from '@angular/core/testing';

import { BrowserModule } from '@angular/platform-browser';
import { NgModule } from '@angular/core';
import { FormsModule } from '@angular/forms';
import { AppComponent } from './app.component';
import {FilterPipe, SortByPipe} from './pipes';

import DynamicSerializer from 'dynamic-serializer';


describe('AppComponent', () => {
  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [
        AppComponent,
        FilterPipe, 
        SortByPipe
      ],
      imports: [
    BrowserModule,
    FormsModule
  ],
    }).compileComponents();
  }));

  it('should create the app', async(() => {
    const fixture = TestBed.createComponent(AppComponent);
    const app = fixture.debugElement.componentInstance;
    expect(app).toBeTruthy();
  }));

  it(`should have as title 'app'`, async(() => {
    const fixture = TestBed.createComponent(AppComponent);
    const app = fixture.debugElement.componentInstance;
    expect(app.title).toEqual('app');
  }));

  it('should render title in a h1 tag', async(() => {
    const fixture = TestBed.createComponent(AppComponent);
    fixture.detectChanges();
    const compiled = fixture.debugElement.nativeElement;
    expect(compiled.querySelector('h1').textContent).toContain('Welcome to app!');
  }));

  //Component test
  it('should match snapshots', async(() => {
    const fixture = TestBed.createComponent(AppComponent);
    const app = fixture.debugElement.componentInstance;
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  }));

  // function test
  it('gives proper change.', () => {
    const fixture = TestBed.createComponent(AppComponent);
    const app = fixture.debugElement.componentInstance;
    expect(app.changeMaker(1.99)).toMatchSnapshot();
    expect(app.changeMaker(2.75)).toMatchSnapshot();
    expect(app.changeMaker(0.24)).toMatchSnapshot();
  })

	// on dynamic field
	it('makes simple dynamic fields deterministic', () => {
		const dynamicSerializer = new DynamicSerializer();
		const fullPaths = ['primaryKey', 'wholeArray', 'singleValue.levelOne.arr.3'];
		const userId = Math.random();
		const results = {
			staticProp: 'Hi there',
			primaryKey: userId,
			wholeArray: [Math.random(), Math.random()],
			singleValue: {
			levelOne: {
				arr: ['badger', 'badger', 'mushroom', 'snake' + Math.random()]
			}
		}
		};
		const moreResults = {
			foreignKey: userId
		};
		dynamicSerializer.toStatic(results, ['primaryKey', 'wholeArray', 'singleValue.levelOne.arr.3']);
		dynamicSerializer.toStatic(moreResults, ['foreignKey']);

		expect(results).toMatchSnapshot();
		expect(moreResults).toMatchSnapshot();
		expect(results.primaryKey).toBe(moreResults.foreignKey);
	})

});
