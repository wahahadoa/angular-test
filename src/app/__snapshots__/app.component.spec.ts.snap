// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent gives proper change. 1`] = `
Object {
  "dimes": 2,
  "nickels": 0,
  "pennies": 5,
  "quarters": 7,
}
`;

exports[`AppComponent gives proper change. 2`] = `
Object {
  "dimes": 0,
  "nickels": 0,
  "pennies": 1,
  "quarters": 11,
}
`;

exports[`AppComponent gives proper change. 3`] = `
Object {
  "dimes": 2,
  "nickels": 0,
  "pennies": 5,
  "quarters": 0,
}
`;

exports[`AppComponent makes simple dynamic fields deterministic 1`] = `
Object {
  "primaryKey": 0,
  "singleValue": Object {
    "levelOne": Object {
      "arr": Array [
        "badger",
        "badger",
        "mushroom",
        3,
      ],
    },
  },
  "staticProp": "Hi there",
  "wholeArray": Array [
    1,
    2,
  ],
}
`;

exports[`AppComponent makes simple dynamic fields deterministic 2`] = `
Object {
  "foreignKey": 0,
}
`;

exports[`AppComponent should match snapshots 1`] = `
<app-root
  songs={[Function Array]}
  term={[Function Number]}
  title={[Function String]}
>
  <div
    style="text-align:center"
  >
    <h1>
      Welcome to app!
    </h1>
    <img
      src="data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0idXRmLTgiPz4NCjwhLS0gR2VuZXJhdG9yOiBBZG9iZSBJbGx1c3RyYXRvciAxOS4xLjAsIFNWRyBFeHBvcnQgUGx1Zy1JbiAuIFNWRyBWZXJzaW9uOiA2LjAwIEJ1aWxkIDApICAtLT4NCjxzdmcgdmVyc2lvbj0iMS4xIiBpZD0iTGF5ZXJfMSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB4bWxuczp4bGluaz0iaHR0cDovL3d3dy53My5vcmcvMTk5OS94bGluayIgeD0iMHB4IiB5PSIwcHgiDQoJIHZpZXdCb3g9IjAgMCAyNTAgMjUwIiBzdHlsZT0iZW5hYmxlLWJhY2tncm91bmQ6bmV3IDAgMCAyNTAgMjUwOyIgeG1sOnNwYWNlPSJwcmVzZXJ2ZSI+DQo8c3R5bGUgdHlwZT0idGV4dC9jc3MiPg0KCS5zdDB7ZmlsbDojREQwMDMxO30NCgkuc3Qxe2ZpbGw6I0MzMDAyRjt9DQoJLnN0MntmaWxsOiNGRkZGRkY7fQ0KPC9zdHlsZT4NCjxnPg0KCTxwb2x5Z29uIGNsYXNzPSJzdDAiIHBvaW50cz0iMTI1LDMwIDEyNSwzMCAxMjUsMzAgMzEuOSw2My4yIDQ2LjEsMTg2LjMgMTI1LDIzMCAxMjUsMjMwIDEyNSwyMzAgMjAzLjksMTg2LjMgMjE4LjEsNjMuMiAJIi8+DQoJPHBvbHlnb24gY2xhc3M9InN0MSIgcG9pbnRzPSIxMjUsMzAgMTI1LDUyLjIgMTI1LDUyLjEgMTI1LDE1My40IDEyNSwxNTMuNCAxMjUsMjMwIDEyNSwyMzAgMjAzLjksMTg2LjMgMjE4LjEsNjMuMiAxMjUsMzAgCSIvPg0KCTxwYXRoIGNsYXNzPSJzdDIiIGQ9Ik0xMjUsNTIuMUw2Ni44LDE4Mi42aDBoMjEuN2gwbDExLjctMjkuMmg0OS40bDExLjcsMjkuMmgwaDIxLjdoMEwxMjUsNTIuMUwxMjUsNTIuMUwxMjUsNTIuMUwxMjUsNTIuMQ0KCQlMMTI1LDUyLjF6IE0xNDIsMTM1LjRIMTA4bDE3LTQwLjlMMTQyLDEzNS40eiIvPg0KPC9nPg0KPC9zdmc+DQo="
      width="100"
    />
  </div><div
    style="margin: 0 auto;width: 600px;"
  >
    <div>
      test Me now !!
    </div>
    <input
      name="term"
      ng-reflect-name="term"
      ng-reflect-model="5"
      class="ng-untouched ng-pristine ng-valid"
    />
    <table>
      <tbody>
        <tr>
          <td>
            Song 5
          </td>
          <td>
            50
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</app-root>
`;
